### PR TITLE
Z: Remove TR_USEDUALTLH env var

### PR DESCRIPTION
**Pull request by ehsankianifar**
The TR_USEDUALTLH was introduced to block non zero init in TLH until performance analysis was done.
This never meant to be permanent and this PR removes it.

### CODE REVIEW

**Comment by ehsankianifar** [comment]
@r30shah FYI

File: runtime/compiler/z/codegen/J9TreeEvaluator.cpp [hunk 1]
```diff
@@ -10135,9 +10133,11 @@ genHeapAlloc(TR::Node * node, TR::Instruction *& iCursor, bool isVariableLen, TR
             }
          }
 
-      if (!comp->getOption(TR_DisableDualTLH) && useDualTLH && node->canSkipZeroInitialization())
+      if (!comp->getOption(TR_DisableDualTLH) && node->canSkipZeroInitialization())
          {
-               iCursor = generateRXInstruction(cg, TR::InstOpCode::getCmpLogicalOpCode(), node, sizeReg,
+         cg->generateDebugCounter(TR::DebugCounter::debugCounterName(comp, "dualTLH/nonZeroAlloc/{%s}{%s}", comp->signature(),comp->getHotnessName(comp->getMethodHotness())),1,TR::DebugCounter::Undetermined);
```
Comment (ehsankianifar) [inline runtime/compiler/z/codegen/J9TreeEvaluator.cpp:10138]: make sure there is no branching before this to other code path that allocate heap!

Comment (ehsankianifar) [reply runtime/compiler/z/codegen/J9TreeEvaluator.cpp:10138]: move it to line 10153 to not count if calling helper

**Comment by ehsankianifar** [comment]
Java 8 build successful on zLinux and sanity tests pass! (Build ID: 67422)
Java 17 build successful on zLinux
@r30shah

**Review by r30shah — APPROVED** [review]

**Comment by r30shah** [comment]
Jenkins test sanity zlinux64 jdk8,jdk11,jdk17

**Comment by r30shah** [comment]
Jenkins test sanity zlinux jdk8,jdk11,jdk17

**Comment by r30shah** [comment]
zLinux test passes and the PR removes the use of environment variable to enable DualTLH (Should use the available JIT option to enable that). Merging the change.

**Pull request merged**